salt/kubernetes-master/init.sls

```diff
@@ -64,6 +64,20 @@ kube-apiserver:
       - file:     kube-apiserver
       - sls:      ca-cert
       - sls:      cert
+  # We need to wait for Kube API server to actually start, see k8s issue #47739
+  # TODO: Salt doesn't seem to have a retry mechanism in the version were using,
+  # so I'm doing a horrible hack right now.
```

Comment:
I use this hack all over the place when I interact with kubectl. I don't like it, but... not a lot I can do. :(